packages/lesswrong/components/tagging/SubforumNotificationSettings.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from "react";
+import { Components, getFragment, registerComponent } from "../../lib/vulcan-lib";
+import NotificationsNoneIcon from "@material-ui/icons/NotificationsNone";
+import NotificationsIcon from "@material-ui/icons/Notifications";
+import { AnalyticsContext } from "../../lib/analyticsEvents";
+import IconButton from "@material-ui/core/IconButton";
+import ClickAwayListener from "@material-ui/core/ClickAwayListener";
+import Paper from "@material-ui/core/Paper";
+import UserTagRels from "../../lib/collections/userTagRels/collection";
+import { useMulti } from "../../lib/crud/withMulti";
+import { Link } from "../../lib/reactRouterWrapper";
+
+const styles = (theme: ThemeType): JssStyles => ({
+  notificationsButton: {
+    margin: "0px 12px 10px 0px",
+    padding: 4,
+  },
+  popout: {
+    padding: "1px 0px 4px 0px",
+  },
+  accountLink: {
+    marginTop: -6,
+    padding: "0px 8px",
+    fontStyle: "italic",
+  },
+});
+
+const SubforumNotificationSettings = ({
+  tag,
+  currentUser,
+  classes,
+}: {
+  tag: TagBasicInfo;
+  currentUser: UsersCurrent;
+  classes: ClassesType;
+}) => {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const [open, setOpen] = useState(false);
+
+  const { LWPopper, WrappedSmartForm, Typography, Loading } = Components;
+
+  const { loading, results } = useMulti({
+    terms: { view: "single", tagId: tag._id, userId: currentUser._id },
+    collectionName: "UserTagRels",
+    fragmentName: "UserTagRelNotifications",
+    fetchPolicy: "cache-and-network",
+  });
+
+  // Don't show notification settings if the user is not subscribed to the tag
+  if (!currentUser || !currentUser.profileTagIds.includes(tag._id)) return null;
+
+  const userTagRel = results?.length ? results[0] : undefined;
+
+  const handleOpen = (event) => {
+    setOpen(true);
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = (e) => {
+    if (e && anchorEl?.contains(e.target)) {
+      return;
+    }
+    setOpen(false);
+    setAnchorEl(null);
+  };
+
+  const handleToggle = (e) => {
+    if (open) {
+      handleClose(null); // When closing from toggle, force a close by not providing an event
+    } else {
+      handleOpen(e);
+    }
+  };
```

Comment:
I feel like there should be a better way to handle this open/close logic, though I don't have a specific suggestion as of now. At least I think there should be comments in the code explaining what's happening. I'll try to play around with this more, and/or we could pair on it. :)

Also I'm confused why we need to set and unset `anchorEl`? That doesn't change right?